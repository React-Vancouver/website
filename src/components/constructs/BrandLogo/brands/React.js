import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({ className }) => {
    return (
        <svg
            className={className}
            fill="none"
            viewBox="0 0 32 29"
            xmlns="http://www.w3.org/2000/svg"
        >
            <path
                fill="#61DAFB"
                d="M32 14.228c0-2.117-2.655-4.122-6.725-5.366.94-4.141.522-7.436-1.317-8.49-.424-.248-.92-.365-1.461-.365v1.452c.3 0 .54.058.743.169.887.508 1.272 2.442.972 4.93a20.03 20.03 0 01-.333 1.913 31.658 31.658 0 00-4.142-.71 31.794 31.794 0 00-2.713-3.255c2.126-1.973 4.122-3.054 5.48-3.054V0c-1.795 0-4.143 1.276-6.517 3.49C13.613 1.29 11.265.026 9.471.026v1.452c1.35 0 3.352 1.075 5.479 3.034a30.762 30.762 0 00-2.694 3.25c-1.474.156-2.87.397-4.148.716-.15-.651-.261-1.283-.34-1.888-.306-2.488.072-4.422.953-4.936.195-.117.45-.17.75-.17V.034c-.548 0-1.044.117-1.474.364C6.164 1.452 5.753 4.74 6.699 8.87 2.642 10.119 0 12.118 0 14.228c0 2.116 2.655 4.121 6.725 5.365-.94 4.141-.522 7.436 1.317 8.491.424.247.92.365 1.468.365 1.794 0 4.142-1.277 6.516-3.49 2.374 2.2 4.723 3.464 6.516 3.464.548 0 1.044-.118 1.474-.365 1.833-1.055 2.244-4.343 1.298-8.471 4.044-1.244 6.686-3.25 6.686-5.36zm-8.492-4.344a29.28 29.28 0 01-.881 2.572 35.198 35.198 0 00-1.794-3.086c.927.137 1.82.306 2.675.514zM20.52 16.82a34.66 34.66 0 01-1.572 2.488 33.975 33.975 0 01-5.883.006 33.919 33.919 0 01-2.935-5.066 34.697 34.697 0 012.922-5.085 33.995 33.995 0 015.883-.007 33.919 33.919 0 012.935 5.066 36.471 36.471 0 01-1.35 2.598zm2.107-.846c.352.872.652 1.745.9 2.591-.854.209-1.755.384-2.687.521a35.832 35.832 0 001.787-3.112zm-6.614 6.947a26.83 26.83 0 01-1.813-2.083c.587.026 1.187.045 1.793.045.614 0 1.22-.013 1.814-.045a25.428 25.428 0 01-1.794 2.083zm-4.853-3.835a29.452 29.452 0 01-2.674-.514c.241-.84.541-1.706.88-2.572.268.52.548 1.042.855 1.562.306.521.62 1.03.94 1.524zm4.82-13.55a26.863 26.863 0 011.814 2.083A40.438 40.438 0 0016 7.573c-.613 0-1.22.013-1.813.045a25.437 25.437 0 011.793-2.083zM11.154 9.37a36.045 36.045 0 00-1.788 3.106 27.871 27.871 0 01-.9-2.592 31.73 31.73 0 012.688-.514zM5.25 17.522c-2.31-.983-3.803-2.272-3.803-3.294 0-1.023 1.494-2.319 3.803-3.295a20.112 20.112 0 011.806-.658 31.237 31.237 0 001.468 3.966 30.797 30.797 0 00-1.448 3.946 19.857 19.857 0 01-1.826-.665zm3.509 9.305c-.887-.508-1.272-2.442-.972-4.929.072-.612.19-1.257.333-1.914 1.278.312 2.674.553 4.141.71a31.797 31.797 0 002.714 3.255c-2.126 1.973-4.122 3.054-5.48 3.054-.293-.006-.54-.065-.736-.176zm15.472-4.962c.306 2.488-.072 4.422-.953 4.936-.195.118-.45.17-.75.17-1.35 0-3.352-1.075-5.479-3.035a30.765 30.765 0 002.694-3.249 30.507 30.507 0 004.148-.716c.15.657.268 1.289.34 1.895zm2.51-4.343c-.56.241-1.173.456-1.806.658a31.244 31.244 0 00-1.468-3.965 30.789 30.789 0 001.448-3.946c.646.201 1.26.423 1.833.664 2.31.983 3.803 2.272 3.803 3.295-.006 1.022-1.5 2.318-3.81 3.294z"
            ></path>
            <path
                fill="#61DAFB"
                d="M15.993 17.203a2.978 2.978 0 002.981-2.975 2.978 2.978 0 00-2.98-2.976 2.978 2.978 0 00-2.981 2.976 2.978 2.978 0 002.98 2.975z"
            ></path>
        </svg>
    );
};

Icon.propTypes = {
    className: PropTypes.string,
};

export default Icon;
