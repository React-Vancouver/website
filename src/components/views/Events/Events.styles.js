import S from '@symbols';

export const rootClasses = {
  minHeight: `calc(100vh - ${S.LAYOUT_FOOTER_HEIGHT} - ${S.LAYOUT_NAV_HEIGHT})`,
};

export const heroStyles = {
  display: 'grid',
  gridTemplateColumns: 'repeat(2, 1fr)',
  gap: S.calcSpace(3),
  [`@media (max-width: ${S.LAYOUT_MOBILE_MAX})`]: {
    gridTemplateColumns: '1fr',
    gap: 0,
  },
};

export const timelineContainerStyles = {
  borderLeft: S.LINE_DEFAULT_BORDER,
  [`@media (max-width: ${S.LAYOUT_MOBILE_MAX})`]: {
    borderLeft: 'none',
  },
};

export const timelineMarkStyles = {
  height: S.calcSpace(2, 0.1),
  width: S.calcSpace(2, 0.1),
  border: S.LINE_DEFAULT_BORDER,
  backgroundColor: S.COLOR_GREYS.WHITE,
  borderRadius: '50%',
  position: 'absolute',
  left: S.calcSpace(-1, -0.1),
  top: S.calcSpace(2, -0.1),
  [`@media (max-width: ${S.LAYOUT_MOBILE_MAX})`]: {
    display: 'none',
  },
};

export const eventStyles = {
  position: 'relative',
  display: 'grid',
  gridTemplateColumns: '1fr 3fr',
  gap: S.calcSpace(3),
  [`@media (max-width: ${S.LAYOUT_MOBILE_MAX})`]: {
    gridTemplateColumns: '1fr',
  },
};
