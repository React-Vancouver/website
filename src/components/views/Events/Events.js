import React from 'react';
import PropTypes from 'prop-types';
import { withSpacing } from '@utilities/styles/spacing';
import EventCard from '@constructs/EventCard';
import Button from '@elements/Button';
import { format } from 'date-fns'

import {
  heroStyles,
  timelineContainerStyles,
  timelineMarkStyles,
  timestampStyles,
  eventStyles,
} from './Events.styles';

import fakeEvent from '../../../../__mocks__/data/event-mock';

const eventsData = fakeEvent({ withTalks: true });

import Text from '@elements/Text';
import Line from '@elements/Line';
import Box from '@elements/Box';

const Events = ({ className }) => {
  return (
    <Box className={className}>
      <Box container>
        <Box
          px4
          pb4
          pt8
          my8
          backgroundColor="grey_30"
          corner="lg"
          css={heroStyles}
        >
          <Text element="h1" title scale={20} color="secondary_l">
            Events
          </Text>
          <Box>
            <Text element="h2" mt2 mb8 subheading color="white">
              Monthly gatherings of engineers of all levels. Get inspired, meet your peers and get connected to opportunities.
            </Text>
            <Button mb4>Check our next event</Button>
          </Box>
        </Box>
      </Box>
      <Line />
      <Box container py4 css={timelineContainerStyles}>
        {Array.from(Array(6)).map((_, i) => (
          <Box css={eventStyles} key={i} mb4>
            <div css={timelineMarkStyles} />
            <Text mt2 ml2 caption css={timestampStyles}>{format(eventsData.date, 'do MMMM, y')}</Text>
            <EventCard ml4 data={eventsData} />
          </Box>
        ))}
      </Box>
    </Box>
  );
};

Events.propTypes = {
  className: PropTypes.string,
};

export default withSpacing(Events);
