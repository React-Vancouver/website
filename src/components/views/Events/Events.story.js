import React from 'react';
import { storiesOf } from '@storybook/react';
import Events from '@views/Events';
import notes from './Events.notes.md';

storiesOf('Views', module).add(
    'Events',
    () => (
        <Events />
    ),
    { notes }
);