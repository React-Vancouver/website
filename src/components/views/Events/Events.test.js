import React from 'react';
import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";

import Events from './Events';

describe('Events', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Events />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
});