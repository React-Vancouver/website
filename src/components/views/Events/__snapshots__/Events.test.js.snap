// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders correctly 1`] = `
<div
  className="css-179updh-Events"
>
  <div
    className="css-1yaaf9v-Box"
  >
    <span
      className="css-16ln220-Text"
    >
      Events
    </span>
  </div>
</div>
`;
